lib/screens/movie_list.dart

```diff
@@ -0,0 +1,110 @@
+import 'package:flutter/material.dart';
+import '../widgets/common/stroked_text.dart';
+import '../widgets/movie_list/list_posters.dart';
+import '../../repository/interface/movie_interface.dart';
+import '../../repository/movie_repo.dart';
+import '../constants/constants.dart';
+import '../model/movie_data.dart';
+import './movie_details.dart';
+
+class MovieListScreen extends StatefulWidget {
+  const MovieListScreen({super.key});
+
+  @override
+  State<StatefulWidget> createState() => _MovieListScreenState();
+
+  static const String routeName = "/";
+}
+
+class _MovieListScreenState extends State<MovieListScreen> {
+  late final Future<List<Movie>> data;
+  final MovieInterface movieRepo = MovieRepository();
+
+  @override
+  void initState() {
+    super.initState();
+    data = getUiData();
+  }
+
+  Future<List<Movie>> getUiData() async {
+    List<Movie> movies = await movieRepo.getMovies();
+    return movies;
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      backgroundColor: Constants.c1,
+      body: SafeArea(
+        child: FutureBuilder<List<Movie>>(
+          future: data,
+          builder: (
+            BuildContext context,
+            AsyncSnapshot<List<Movie>> snapshot,
+          ) {
+            if (snapshot.connectionState == ConnectionState.waiting) {
+              return const CircularProgressIndicator();
+            } else if (snapshot.connectionState == ConnectionState.done) {
+              if (snapshot.hasError) {
+                return const Text(Constants.snapshotError);
+              } else if (snapshot.hasData) {
+                return ListView.builder(
+                  itemCount: snapshot.data?.length,
+                  itemBuilder: (
+                    BuildContext context,
+                    int index,
+                  ) {
+                    final payload = snapshot.data![index];
+                    return GestureDetector(
+                      onTap: () {
+                        Navigator.pushNamed(
+                          context,
+                          DetailScreen.routeName,
+                          arguments: payload,
+                        );
+                      },
+                      child: Card(
+                        color: Constants.c2,
```

Comment:
You could extract this as a separate widget, cleaning this build method

Reply:
Done